### ISSUE TITLE
Is the random traffic script open source for users?

### ISSUE THREAD

**Issue by JasonBourne1998** [lgsvl/simulator#1979]
When my classmates and I build a scene in the pythonAPI and connect with Apollo's dreamview, activating prediction, performance and traffic light will occupy a lot of GPU memory. However, when we use random traffic to control the car and connect it with dreamiew, it will be much easier to load the model. Therefore, we want to modify the scene directly on the basis of random traffic. Is the script of random traffic open source?

**Comment by lemketron** [comment]
Random Traffic was implemented directly in SVL Simulator, but recently we implemented it using Python API. 

You can find the Python script for Random Traffic in the (new) "runner" repo here: 
https://github.com/lgsvl/runner/blob/master/random-traffic-runner/scenario_runner/run_random_traffic.py

Note that this was implemented with a specific future use case in mind and may or may not work as-is with SVL Simulator 2021.3 (the latest public release).

**Comment by JasonBourne1998** [comment]
> 

Thanks for your reply, now we have run it successfully, but now we want to join pedestrians who are crossing the road, but it seems that lgsvl does not provide the insertion of pedestrians who cross the road obey the traffic rules. Now it only supports walking randomly on the crosswalk and walking according to the waypoint. However, as we all know, when pedestrians cross the road, they need to cross according to the red street lights. Does lgsvl have a script in this regard? If not, I wonder if I can use randomly inserted NPC cars to transform the script to create pedestrians who comply with traffic regulations? (because the cars randomly inserted in lgsvl obey the traffic rules), is this feasible?

**Comment by JasonBourne1998** [comment]
> Random Traffic was implemented directly in SVL Simulator, but recently we implemented it using Python API.
> 
> You can find the Python script for Random Traffic in the (new) "runner" repo here: https://github.com/lgsvl/runner/blob/master/random-traffic-runner/scenario_runner/run_random_traffic.py
> 
> Note that this was implemented with a specific future use case in mind and may or may not work as-is with SVL Simulator 2021.3 (the latest public release).

Thanks for your reply, now we have run it successfully, but now we want to join pedestrians who are crossing the road, but it seems that lgsvl does not provide the insertion of pedestrians who cross the road obey the traffic rules. Now it only supports walking randomly on the crosswalk and walking according to the waypoint. However, as we all know, when pedestrians cross the road, they need to cross according to the red street lights. Does lgsvl have a script in this regard? If not, I wonder if I can use randomly inserted NPC cars to transform the script to create pedestrians who comply with traffic regulations? (because the cars randomly inserted in lgsvl obey the traffic rules), is this feasible?

**Comment by EricBoiseLGSVL** [comment]
Yes, you will need to alter pedestrian control classes and may need to add pedestrian annotation data (stoplines, etc).

**Comment by JasonBourne1998** [comment]
> 



> Yes, you will need to alter pedestrian control classes and may need to add pedestrian annotation data (stoplines, etc).

Thank you very much for your reply. I just conducted an experiment based on your reply, but unfortunately, there is no borregasave's unity file in the build interface of unity simulator page, which means that I can't add pedestrians directly in the map. In view of the fact that we will use Apollo for simulation detection in the end, which means that we must do all these operations on borregasave. I checked the previous issue, In 2019, you replied that borregasave's unity was not open source. Is there an update now? If we can't operate on the unity, is there another way for us to deal with this problem?

**Comment by EricBoiseLGSVL** [comment]
Yes, Borregas Ave is not open source.  If you can't make a map to edit, then you will need to use the annotation provided.  Have you used the VSE editor?  Maybe you can annotate paths and then edit the api pathing to randomize and follow pedestrian rules.  The best bet is to make annotation edits to a new map and create new annotation data/logic to succeed.
